contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -4052,6 +4147,79 @@ TdsSendTypeDatetimeoffset(FmgrInfo *finfo, Datum value, void *vMetaData)
 	return rc;
 }
 
+int 
+TdsSendSpatialHelper(FmgrInfo *finfo, Datum value, void *vMetaData, int TdsInstr)
+{
+    int    rc = EOF,
+           npoints,
+           len,             /* number of bytes used to store the string. */      
+           actualLen;       /* Number of bytes that would be needed to
+                             * store given string in given encoding. */
+    char   *destBuf, 
+           *buf, 
+           *itr;
+
+	int32_t   srid;
+
+    TdsColumnMetaData *col = (TdsColumnMetaData *) vMetaData;
+    GSERIALIZED *gser;          /* Used to Store the bytes in the Format which is stored in PostGIS */
+
+    gser = (GSERIALIZED *)PG_DETOAST_DATUM(value);
+    npoints = *((int *)gser->data);
+    /*
+     * Row chunck length expected by the driver is:
+     * 16 * (No. of Points) + 6
+     * 16 -> 2 8-Byte float coordinates (TODO: Need to change when Z and M flags are defined for N-dimension Points)
+     * 6 -> 4 Byte SRID + 2 Byte (01 0C)
+    */
+    len = npoints*16 + 6;      
+    buf = (char *) palloc0(len);
```

Comment:
freeing the buffer?

Reply:
Actually Freeing Buf is causing server crash, So I don't think we need to free buf